common4j/src/main/com/microsoft/identity/common/java/net/SSLSocketFactoryWrapper.java

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.net;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import javax.net.ssl.HandshakeCompletedEvent;
+import javax.net.ssl.HandshakeCompletedListener;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.Synchronized;
+import lombok.experimental.Accessors;
+
+/**
+ * This class is a SSLSocketFactory wrapper that supports Higher TLS by default.
+ * In Android, the default socket would return one that only supports up to TLSv1.1 if API<20
+ * reference: https://developer.android.com/reference/javax/net/ssl/SSLSocket
+ */
+public class SSLSocketFactoryWrapper extends SSLSocketFactory {
+    private static final String TAG = SSLSocketFactoryWrapper.class.getSimpleName();
+
+    /**
+     * SSL Protocols that our library supports.
+     */
+    public static final List<String> SUPPORTED_SSL_PROTOCOLS =
+            Collections.unmodifiableList(Arrays.asList("SSLv2Hello", "SSLv3", "TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3"));
+
+    // Gets TLS version of the latest-established socket connection. For testing only.
+    // NOTE: This onMethod thing doesn't generate javadoc, but this method is only exposed for testing only.
+    @Getter(value = AccessLevel.PACKAGE, onMethod_={@Synchronized})
+    @Setter(value = AccessLevel.PACKAGE, onMethod_={@Synchronized})
```

Comment:
we already discussed not to use this `onMethod` notation: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/1360

Reply:
@shahzaibj This is exposed for test case only (as mentioned in the comment above).

Reply:
This is not in test directory though...so my understanding is that an attempt will be made to generate javadoc for this class and it will fail. Are you doing anything special to not generate javadoc for these?